canonical-schema.json

```diff
@@ -120,10 +120,17 @@
           , "pattern"    : "^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$"
           },
 
-      "optional":
-          { "description": "An identifier for similar optional test cases (kebab-case)"
-          , "type"       : "string"
-          , "pattern"    : "^[a-z]+(-[a-z]+)*$"
+      "scenario":
+          { "description": "An identifier for a specific scenario (kebab-case)"          
+          , "type": "string"
+          , "enum": ["big-integer", "floating-point", "library-test", "unicode"]
```

Comment:
I've taken the liberty of adding the actual values here, which makes the JSON schema validation a lot nicer.

Reply:
This seems easy to miss when adding a new scenario. Could we somehow merge both lists?

Reply:
I thought of maybe having a CI check to verify that the two match? The benefit of the file is that is easier to read. This has the benefit of actually verifying the actual scenario values.

Reply:
But a check works too, this isn't blocking

Reply:
> Can we regenerate the schema when the file changes? 

We could. Good idea.